src/global/trampoline.rs

```diff
@@ -276,12 +277,18 @@ impl Trampoline {
     /// Write the manifest file of the trampoline
     async fn write_manifest(&self) -> miette::Result<()> {
         let manifest_string = serde_json::to_string_pretty(&self.metadata).into_diagnostic()?;
-        tokio::fs::write(self.manifest_path(), manifest_string)
+        tokio_fs::write(self.manifest_path(), manifest_string)
             .await
             .into_diagnostic()?;
 
         Ok(())
     }
+
+    /// Check if binary is a saved trampoline
+    pub async fn is_trampoline(path: &Path) -> miette::Result<bool> {
+        let bytes = tokio_fs::read(path).await.into_diagnostic()?;
+        Ok(bytes == TRAMPOLINE_BIN)
+    }
```

Comment:
How about you check first whether the file has the same number of bytes as the trampoline? Reading whole files which might be many megabytes big could be expensive.

Reply:
changed how I check it